android/src/main/java/de/simolus3/fluttie/FluttieAnimation.java

```
@@ -39,6 +39,7 @@ public FluttieAnimation(FluttiePlugin plugin, TextureRegistry.SurfaceTextureEntr
 				(int) (scale * bounds.width()), (int) (scale * bounds.height()));
 
 		drawable = new LottieDrawable();
+		drawable.setCallback(plugin.getAnimationCallback());
```

Can you briefly help me understand why this `callback` is necessary now? Is it due to some changes in the updated Lottie version or should it have always been like this? Thanks!

---

I think it should have always been like that.

If you look through https://github.com/airbnb/lottie-android/blob/master/lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java, there's a lot of usages of getCallback() that will just be NOPs if this call is not made.

Personally, I wasted quite some time on this little silent failure: https://github.com/airbnb/lottie-android/blob/master/lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java#L846. Had that if statement not been there, at least this warning would have been written: https://github.com/airbnb/lottie-android/blob/master/lottie/src/main/java/com/airbnb/lottie/manager/ImageAssetManager.java#L39.